include/datadog/config.h

```diff
@@ -56,14 +62,72 @@ struct ConfigMetadata {
       : name(n), value(std::move(v)), origin(orig), error(std::move(err)) {}
 };
 
+// Return the chosen configuration value, in order of precedence: `from_env`,
+// `from_user`, and `fallback`.
+// This overload directly populates both telemetry_configs[config_name] and
+// metadata[config_name] using the stringified value, with all values found,
+// from lowest to highest precedence. Returns the chosen value directly.
+//
+// The fallback parameter is optional and defaults to nullptr.
+template <typename Value, typename Stringifier = std::nullptr_t,
+          typename DefaultValue = std::nullptr_t>
+Value pick(const Optional<Value>& from_env, const Optional<Value>& from_user,
```

Comment:
well, `pick` is not _just_ picking a value from a list. Either create the whole chain of config differently or find a better name to this function.